scripts/fn_paradrop.sqf

```diff
@@ -0,0 +1,213 @@
+ * increasing flyInHeight before using it.  Also, don't try with helicopters.
+ *
+ * Optional Arguments: Reset Height
+ * For drop purposes flyInHeight is set during the run.  If the transport is to be
+ * otherwise used after the run, set this to <true> in order to resume normal behavior
+ *
+ * Optional Arguments: Disable Waypoint 2
+ * Occasionally, AI behavior can be wonky especially when drop WPs are around airports.
+ * This will cause them to veer off from their drop direction and attempt to land. 
+ * Disabling WP 2 stops this behavior from occurring on the second waypoint.
+ * If it occurs on the first, I recommend changing the DZ area, the drop length,
+ * or the approach direction. 
+ *
+ * Arguments
+ * 0: Pilot Group <GROUP>
+ * 1: Drop Groups <ARRAY>
+ * 2: Drop Target <MARKER>
+ * 3: Drop Mode <ARRAY> or <NUMBER> (default: 0)
+ * 4: Reset Height <BOOL> (default: false) 
+ * 5: Disable Waypoint 2 <BOOL> (default: false)
+ *
+ * Return Value:
+ * None
+ *
+ * Examples:
+ * [pilot_group, [airborne_1], "drop_marker"] call TAC_Scripts_fnc_paradrop;
+ * [pilot_group, [airborne_1,airborne_2], "drop_marker"] call TAC_Scripts_fnc_paradrop;
+ * [pilot_group, [airborne_1], "drop_marker", 1] call TAC_Scripts_fnc_paradrop;
+ * [pilot_group, [airborne_1], "drop_marker", 0, false, false] call TAC_Scripts_fnc_paradrop;
+ * [pilot_group, [airborne_1], "drop_marker", [200,1000,150]] call TAC_Scripts_fnc_paradrop;
+ */
+
+
+#define DROP_MODES [[150,1000,140],[105,1000,80],[250,1000,140],[4500,1000,900],[3000,1000,900]]
+
+params ["_pilotGroup", "_dropGroups", "_loc", ["_dropMode", 0], ["_resetHeight", false], ["_disableWP2", false]];
+
+if (_loc isEqualTo []) exitWith {};
+
+private _dropData = [];
+// first process dropMode
+if (_dropMode isEqualType 0) then {
+    _dropData = DROP_MODES select _dropMode;
+} else {
+    if (_dropMode isEqualType [] && {count _dropMode == 3}) then {
+        _dropData = _dropMode;
+    };
+};
+
+_dropData params ["_dropHeight", "_dropLength", "_chuteHeight"];
+
+private _vehicle = vehicle (leader _pilotGroup);
+if(isNull _vehicle) exitWith {}; // group not in vehicle
+
+private _origHeight = (getPosATL _vehicle) select 2;
+_vehicle flyInHeight _dropHeight; // set forced flying height
+if (_resetHeight) then {
+    [{
+        params ["_vehicle"];
+        (fullCrew [_vehicle,"cargo"]) isEqualTo []
+    }, {
+        params ["_vehicle", "_origHeight"];
+        _vehicle flyInHeight _origHeight;
+    }, [_vehicle, _origHeight]] call CBA_fnc_waitUntilAndExecute;
+};
+
+// next process waypoints if waypoints desired. 
+// if waypoint is desired, then script will be run on setWaypointStatements
+// if not then script will be run in this function.
+
+// Use addWayPoint for the group piloting the vehicle to add the drop waypoints
+private _dropRunOrigin = getPosATL _vehicle; // calling position from leader will call last position on the ground
```

Comment:
```suggestion
private _dropRunOrigin = getPosATL _vehicle;
```
Comment not accurate, since we switched to `getPosATL` instead of `position` - it would report correctly but delayed. Doing it in the vehicle is always better.

Reply:
Noted, thanks.